src/omv/imlib/filter.c

```diff
@@ -1244,30 +1244,35 @@ void imlib_morph(image_t *img,
                  int offset,
                  bool invert,
                  image_t *mask) {
+    int img_w = img->w, w_limit = img_w - 1;
+    int img_h = img->h, h_limit = img_h - 1;
```

Comment:
```suggestion
```

Reply:
This is done on purpose to move the variable into the local stack so GCC can use one pointer with an offset to load it when needed and/or keep it in the register space. If you don't do this GCC will not necessarily store the dereferenced  value in register space. It may store the pointer to img in register space or the stack, but, then you don't know if it's one or two dereferences to get to w/h.

By doing the above GGC will ensure the variable is stored in the stack frame and it knows it's free to keep in the register space as it likes.

I didn't think this mattered originally until Larry showed me code snippets where the compiler didn't cache the variable like I thought.

Reply:
I understand what this is supposed to be doing, and the compiler probably does that if needed without your intervention, and if it doesn't this doesn't have any significant impact on performance anyway compared to the algorithm. If we follow the same logic then we need to store all variables on stack, unroll all loops and inline all functions, and maybe rewrite all code in asm ? Those micro optimizations are not needed anymore.

Reply:
Changed this, Will stop with these micro optimizations. Did a performance test also:

MicroOptimizations

GRAYSCALE QVGA 8646 us (300 frame average)
GRAYSCALE QQVGA 2220 us (300 frame average)

RGB565 QVGA 54906 us (300 frame average)
RGB565 QQVGA 14454 us (300 frame average)

Regular

GRAYSCALE QVGA 8641 us (300 frame average)
GRAYSCALE QQVGA 2208 us (300 frame average)

RGB565 QVGA 53867 us (300 frame average)
RGB565 QQVGA 14257 us (300 frame average)

Leading two digits of both are the same. The bottom numbers moved around pretty regularly. But, performance is the same.